### PR TITLE
swupd: add tests for packs

### PR DESCRIPTION
**Pull request by cmarcelo**
- Add tests for the logic of figuring out which bundles need delta
  packs between two versions.

- Add tests for creating zero packs, with data from different sources,
  as well as incomplete data (so fallback logic kicks in), validating
  the contents of the generated pack.

- Add testFileSystem helper struct, to make filesystem-like operations
  in a test directory. By using methods of the struct, we reduce a bit
  the noise of passing t and dir around.

Signed-off-by: Caio Marcelo de Oliveira Filho <caio.oliveira@intel.com>

### CODE REVIEW

**Comment by cmarcelo** [comment]
@matthewrsj will add non-zero packs tests together with the delta integration.

File: swupd/packs_test.go [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+			}
+		}
+		toM := &Manifest{}
+		toM.Header.Version = tt.ToV
+		for name, v := range tt.To {
+			addBundle(toM, name, v)
+		}
+
+		bundleMap, err := FindBundlesToPack(fromM, toM)
+		if err != nil {
```
Comment (matthewrsj) [inline swupd/packs_test.go]: I've suggested this elsewhere, you can (optionally) simplify this logic by doing something like
```go
if (err != nil) != tt.ShouldFail {
    if tt.ShouldFail {
        t.Fatalf(...)
    } else {
        t.Fatalf(...)
    }
    continue
}
```

**Comment by cmarcelo** [comment]
> Just an optional suggestion. It would be very nice to use those new helpers in the CreateManifests tests.

Sure, I thought the same, but as the tests were working already, I pushed to later. I'd be happy to take care of it once delta integration is done.

File: swupd/packs_test.go (hunk 1 above)
Comment (cmarcelo) [reply swupd/packs_test.go]: I haven't looked at the BCB review, but I will try that approach there.

Comment (cmarcelo) [reply swupd/packs_test.go]: It did look better on the BCB, so I'll try the same thing here.

Comment (cmarcelo) [reply swupd/packs_test.go]: Done.
